scripts/attach_novaseq_xml.py

```diff
@@ -16,31 +13,85 @@
 """
 
 def main(lims, args):
-    log=[]
-    content = None
     process = Process(lims, id=args.pid)
 
-    # Copy Read and index parameter from the step "Load to Flowcell (NovaSeq 6000 v2.0)"
-    UDF_to_copy = ['Read 1 Cycles', 'Read 2 Cycles', 'Index Read 1', 'Index Read 2']
-    for i in UDF_to_copy:
-        if process.parent_processes()[0].udf.get(i):
-            process.udf[i]=process.parent_processes()[0].udf[i]
-    process.put()
-
-    # Fetch Flowcell ID
-    FCID=process.parent_processes()[0].output_containers()[0].name
-
-    for outart in process.all_outputs():
-        if outart.type == 'ResultFile' and outart.name == 'Run Info':
-            try:
-                lims.upload_new_file(outart,max(glob.glob('/srv/mfs/NovaSeq_data/*{}/RunInfo.xml'.format(FCID)),key=os.path.getctime))
-            except:
-                raise RuntimeError("No RunInfo.xml Found!")
-        elif outart.type == 'ResultFile' and outart.name == 'Run Parameters':
-            try:
-                lims.upload_new_file(outart,max(glob.glob('/srv/mfs/NovaSeq_data/*{}/RunParameters.xml'.format(FCID)),key=os.path.getctime))
-            except:
-                raise RuntimeError("No RunParameters.xml Found!")
+    if "NovaSeqXPlus Run" in process.type.name:
+
+        # Fetch Flowcell ID
+        FCID = process.parent_processes()[0].output_containers()[0].name
+
+        for outart in process.all_outputs():
+            if outart.type == "ResultFile" and outart.name == "Run Info":
+                try:
+                    lims.upload_new_file(
+                        outart,
+                        max(
+                            glob.glob(
+                                "/srv/mfs/NovaSeqXPlus_data/*{}/RunInfo.xml".format(
+                                    FCID
+                                )
+                            ),
+                            key=os.path.getctime,
+                        ),
+                    )
+                except:
```

Comment:
Mostly as a note for the future, try to avoid these catch all try/except. I understand this is just because the old version of the code used it so it's fine to keep for now.